app/models/miq_cockpit_ws_worker.rb

```diff
@@ -0,0 +1,92 @@
+require 'miq_apache'
+require 'runcmd'
+require 'linux_admin'
+
+class MiqCockpitWsWorker < MiqWorker
+  require_nested :Runner
+
+  APACHE_CONF_FILE = '/etc/httpd/conf.d/manageiq-cockpit.conf'.freeze
+  class_attribute(:web_service_uri)
+
+  self.web_service_uri = "http://localhost:4000"
+  self.required_roles = ['cockpit_ws']
+  self.maximum_workers_count = 1
+
+  def friendly_name
+    @friendly_name ||= "Cockpit Worker"
+  end
+
+  def self.can_start_cockpit_ws
+    if @supports_cockpit_ws.nil?
+      @supports_cockpit_ws = MiqEnvironment::Command.is_linux? && \
+                             File.exist?('/usr/libexec/cockpit-ws') && \
+                             File.exist?('/usr/libexec/cockpit-ssh')
+    end
+    @supports_cockpit_ws
+  end
+
+  def self.can_activate_cockpit?(opts = nil)
+    return false unless has_required_role? && MiqServer.my_server.has_active_role?('web_services')
+
+    opts ||= MiqCockpitWsWorker.worker_settings
+    if opts[:external_url]
+      return true
+    end
+    can_start_cockpit_ws
+  end
+
+  # Update settings for other servers
+  def self.update_other_server_configs(opts)
+    url = opts[:external_url]
+    url = MiqCockpit::WS.url(opts, MiqServer.my_server.ipaddress) unless url
+
+    MiqServer.active_miq_servers.where.not(:id => MiqServer.my_server.id) do |server|
+      conf = server.get_config
+      conf[:workers][:cockpit_ws_worker] = {
+        :external_url => url
+      }
+      server.set_config(conf)
+    end
+  end
+
+  def self.update_webserver_url
+    port = MiqWebServiceWorker::STARTING_PORT
+    if MiqWebServiceWorker.registered_ports && !MiqWebServiceWorker.registered_ports.empty?
+      port = MiqWebServiceWorker.registered_ports[0]
+    end
+    uri = MiqWebServiceWorker.build_uri(port)
+    ret = true unless MiqCockpitWsWorker.web_service_uri == uri
+    MiqCockpitWsWorker.web_service_uri = uri
+    ret
+  end
+
+  def self.sync_workers
+    opts = MiqCockpitWsWorker.worker_settings
+    install_apache_proxy_config(opts) if MiqEnvironment::Command.supports_apache?
+    can_activate = can_activate_cockpit?(opts)
+    @workers = opts[:external_url] || can_activate ? 1 : 0
+    needs_restart = update_webserver_url
+    ret = super
+
+    if can_activate
+      update_other_server_configs(opts)
+    end
+
+    # If nothing was added but url changed, restart
+    _log
+    if needs_restart && ret[:adds].empty?
+      MiqCockpitWsWorker.restart_workers
+    end
+    ret
+  end
+
+  def self.install_apache_proxy_config(opts = nil)
+    opts ||= MiqCockpitWsWorker.worker_settings
+    if has_required_role? || opts[:external_url]
+      MiqCockpit::ApacheConfig.new(opts).save(APACHE_CONF_FILE)
+    elsif File.exist?(APACHE_CONF_FILE)
+      File.truncate(APACHE_CONF_FILE, 0)
+    end
+    MiqServer.my_server.queue_restart_apache if MiqEnvironment::Command.supports_apache?
+  end
```

Comment:
@jrafanie Can you review these 3 methods above...I don't trust my judgement on them.  I have a few concerns within them, but I'll withhold until you can review first.